hydra_provider/src/bootstrap.rs

```diff
@@ -0,0 +1,36 @@
+use std::net::SocketAddr;
+
+// use crate::commands::service::CommandService;
+use crate::data_feed_service::serial::{SerialDataFeedService, SerialDataFeedServer};
+use crate::data_feed_service::random::{RandomDataFeedService, RandomDataFeedServer};
+use crate::database_service::DatabaseService;
+
+use tonic::transport::Server;
+use tonic_health::server::health_reporter;
+use tonic_health::pb::health_server::HealthServer;
+use tonic_health::server::HealthService;
+
+// Sets up the gRPC server and loads relevant services
+pub async fn bootstrap(
+	server_port: u32,
+	database_address: String,
+) -> () {
+	let server_address: SocketAddr = format!("[::1]:{}", server_port).parse().unwrap();
+
+	let database_service = DatabaseService::new(&database_address.as_str()).await;
+	let serial_data_feed_service = SerialDataFeedService::new(&database_service);
+	let random_data_feed_service = RandomDataFeedService::new(&database_service);
```

Comment:
So the whole communication thing is happening inside the service. This is just to isolate the "saving" logic from the "receiving " logic. 

Tryina do something like an onion architecture here.

Reply:
> My reply maybe mostly my personal opinions

Yes I understand, but I do think that when we pass the database_service and call `database.save` I think we are making them a bit too coupled

Just to let you know something like this is what I had in mind

![image](https://github.com/uorocketry/rgs/assets/26195439/59c6c851-9a6d-4908-93dc-c353a7db3590)

Here is a more concrete example on how it would look like on the setup

```rs
   let (tx, _) = tokio::sync::broadcast::channel(1024);

    // The producer only knows its going to send stuff somewhere
    let serial_data_feed_service = SerialDataFeedService::new(tx.clone());
    let random_data_feed_service = RandomDataFeedService::new(tx.clone());
    
    
    // The consumer only knows that its going to receive stuff
    let saved_to_database_consumer = SavedToDatabaseConsumer::new(tx.subscribe());
    let stdout_consumer = StdOutConsumer::new(tx.subscribe());
    let my_custom_consumer = CustomConsumer::new(tx.subscribe());
```

Then weather or not the `XFeedService` actually calls the `tx.send` function is up to the `is_running` condition being on. 

For more resilience, inside the database service, when you're processing the messages if by any chance a message failed to be saved we could either have a separate fail-safe requeue (like to a file or to try to send again)
or also plugging in other consumer like one that saves to a text file or just logs it to the console, etc.

Reply:
That's a really good idea but this PR is getting too bloated. I'll add that as a follow up PR.